src/lib/assign/ValueAssign.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/support/Span.h>
+#include <string>
+
+namespace chip {
+namespace Value {
+
+template <class SRC, class DEST>
+SRC & Assign(SRC & dest, const DEST & src)
+{
+    return dest = src;
+}
+
+template <>
+std::string & Assign(std::string & dest, const chip::CharSpan & src)
```

Comment:
This is inside `namespace chip`.  Why does it need `chip::` prefixes here and below?